src/doc/trpl/macros.md

```diff
@@ -653,6 +653,144 @@ macro_rules! bct {
 Exercise: use macros to reduce duplication in the above definition of the
 `bct!` macro.
 
+# Common macros
+
+Here are some common macros you’ll see in Rust code.
+
+## panic!
+
+This macro causes the current thread to panic. You can give it a message
+to panic with:
+
+```rust,ignore
+panic!("oh no!");
+```
+
+## unreachable!
+
+This macro is used when you think some code should never execute:
+
+```rust
+if false {
+    unreachable!();
+}
+```
+
+Sometimes, the compiler may make you have a different branch that you know
+will never, ever run. In these cases, use this macro, so that if you end
+up wrong, you’ll get a `panic!` about it.
+
+```rust
+let x: Option<i32> = None;
+
+match x {
+    Some(_) => unreachable!(),
+    None => println!("I know x is None!"),
+}
+```
+
+## unimplemented!
+
+The `unimplemented!` macro can be used when you’re trying to get your functions
+to typecheck, and don’t want to worry about writing out the body of the
+function.  For example, let’s say we have some program, but it isn’t compiling
+due to a type error, and we can’t figure out why. When we ask someone
+for help, we just care about the types, not what the bodies do, so we can
+give them something like this:
+
+```rust,ignore
+fn foo(i: i32) -> i32 {
+    unimplemented!();
+}
+
+let x: i64 = 100;
+
+foo(x);
+```
```

Comment:
This seems like a bit of an odd example to me, even if `foo` were large here the span for `expected i32, found i64` should point precisely at `foo(x)` and shouldn't have much to do with the implementation of `foo`. I figured that this sort of macro would be used for implementations of traits where you haven't filled out all the required methods yet.